integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/authorization/ChangeImageCatalogTest.java

```diff
@@ -0,0 +1,174 @@
+
+import static com.sequenceiq.it.cloudbreak.context.RunningParameter.expectedMessage;
+
+import javax.inject.Inject;
+import javax.ws.rs.ForbiddenException;
+
+import org.testng.annotations.Test;
+
+import com.sequenceiq.environment.api.v1.environment.model.response.EnvironmentStatus;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.it.cloudbreak.actor.CloudbreakActor;
+import com.sequenceiq.it.cloudbreak.client.CredentialTestClient;
+import com.sequenceiq.it.cloudbreak.client.DistroXTestClient;
+import com.sequenceiq.it.cloudbreak.client.EnvironmentTestClient;
+import com.sequenceiq.it.cloudbreak.client.FreeIpaTestClient;
+import com.sequenceiq.it.cloudbreak.client.ImageCatalogTestClient;
+import com.sequenceiq.it.cloudbreak.client.SdxTestClient;
+import com.sequenceiq.it.cloudbreak.context.Description;
+import com.sequenceiq.it.cloudbreak.context.MockedTestContext;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.credential.CredentialTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.DistroXTestDto;
+import com.sequenceiq.it.cloudbreak.dto.distrox.image.DistroXChangeImageCatalogTestDto;
+import com.sequenceiq.it.cloudbreak.dto.environment.EnvironmentTestDto;
+import com.sequenceiq.it.cloudbreak.dto.freeipa.FreeIpaTestDto;
+import com.sequenceiq.it.cloudbreak.dto.freeipa.FreeipaChangeImageCatalogTestDto;
+import com.sequenceiq.it.cloudbreak.dto.imagecatalog.ImageCatalogTestDto;
+import com.sequenceiq.it.cloudbreak.dto.sdx.SdxChangeImageCatalogTestDto;
+import com.sequenceiq.it.cloudbreak.dto.sdx.SdxInternalTestDto;
+import com.sequenceiq.it.cloudbreak.testcase.AbstractIntegrationTest;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+
+public class ChangeImageCatalogTest extends AbstractIntegrationTest {
+
+    @Inject
+    private EnvironmentTestClient environmentTestClient;
+
+    @Inject
+    private CredentialTestClient credentialTestClient;
+
+    @Inject
+    private CloudbreakActor cloudbreakActor;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private DistroXTestClient distroXClient;
+
+    @Inject
+    private ImageCatalogTestClient imageCatalogTestClient;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        useRealUmsUser(testContext, AuthUserKeys.ACCOUNT_ADMIN);
+        useRealUmsUser(testContext, AuthUserKeys.ENV_CREATOR_A);
+    }
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    @Description(
+            given = "there is a running Freeipa",
+            when = "a change image catalog request is sent",
+            then = "admin user can perform the operation but env creator user should get forbidden exception")
+    public void testChangeFreeipaImageCatalog(MockedTestContext testContext) {
+        useRealUmsUser(testContext, AuthUserKeys.ACCOUNT_ADMIN);
```

Comment:
Would it make sense to do the following:

- env_creator_a creates all resources (image catalog, credential, freeipa, environment...),
- env_creator_b creates a new image catalog,
- env_creator_b tries to change the image catalog to the one he created and it fails,
- account_admin tries to do the same (change the image catalog to env_creator_b's image catalog) and he succeeds.